graph_weather/models/atmorep/model/attention.py

```diff
@@ -0,0 +1,149 @@
+import torch
+import torch.nn as nn
+
+from ..config import AtmoRepConfig
+
+
+class CrossAttention(nn.Module):
+    """
+    Cross-Attention mechanism that enables communication between transformers of different fields.
+    It allows a target tensor (x) to attend to a source tensor (context) and is useful in
+    multi-modal tasks where information from different fields is combined.
+
+    Args:
+        config (AtmoRepConfig): Configuration object containing model parameters like hidden_dim,
+                                 num_heads, and attention_dropout.
+    """
+
+    def __init__(self, config: AtmoRepConfig):
```

Comment:
Could you add the options as arguments, instead of all wrapping it up in a config object? As CrossAttention could be useful for other models as well, it would be a bit more interoperable if it takes arguments instead of just a `config`.